### PR TITLE
Migrated QT doc section to commonmark

### LINKED ISSUE #104971: Convert Qt from Docbook to CommonMark
This issue tracks the conversion of `/doc/languages-frameworks/qt.xml` from Docbook to CommonMark per [RFC 72](https://github.com/NixOS/rfcs/blob/master/rfcs/0072-commonmark-docs.md).

If you intend to work on this conversion, please comment below!

# Instructions for performing this conversion

1. Post in this issue that you intend to work on it. (Please also post if you stop working on it)
2. Checkout a new branch of nixpkgs master.
3. Edit `doc/languages-frameworks/index.xml`: change `qt.xml` to `qt.section.xml`.
4. Move `doc/languages-frameworks/qt.xml` to `doc/languages-frameworks/qt.section.md`.
5. Convert the contents from Docbook to [CommonMark](https://commonmark.org/). Use existing `.md` sections for inspiration.
6. Use `nix-build` from the `doc/` folder to build the manual.
7. Preview the manual on your computer at `http://0.0.0.0:8000/manual.html` after running from `doc/` folder:
  ```
  nix-shell -p python3 --run 'python3 -m http.server --directory ./result/share/doc/nixpkgs'
  ```
8. When it looks good, submit a PR and reference this issue.

### PR DESCRIPTION
**Pull request by wiltaylor**
###### Motivation for this change
Updated QT docs as per Issue #104971

###### Things done
- Converted xml to md and updated index.xml to reference it.

### CODE REVIEW

File: doc/languages-frameworks/qt.section.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+
+Qt
+=
+This section describes the differences between Nix expressions for Qt libraries and applications and Nix expressions for other C++ software. Some knowledge of the latter is assumed.
+
+There are primarily two problems which the Qt infrastructure is designed to address: ensuring consistent versioning of all dependencies and finding dependencies at runtime.
+
+**Nix expression for a Qt package (default.nix)**
```
Comment (jtojnar) [inline doc/languages-frameworks/qt.section.md:8]: Please use actual headings, for headings, not bold paragraphs.

Comment (jtojnar) [inline doc/languages-frameworks/qt.section.md:8]: Also please preserve the anchors:

```suggestion

## Nix expression for a Qt package (default.nix) {#qt-default-nix}
```

File: doc/languages-frameworks/qt.section.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+```nix
+{ mkDerivation, lib, qtbase }: #1
+mkDerivation { #2
```
Comment (jtojnar) [inline doc/languages-frameworks/qt.section.md:12]: Hmm, we will need to figure out the callouts still. Unfortunately, the RFC was more of a declaration so we have to come up with a solution ourselves.

File: doc/languages-frameworks/qt.section.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+> To use another deriver instead of `stdenv.mkDerivation`, use `mkDerivationWith`:
+
+```nix
+mkDerivationWith myDeriver {
+  # ...
+}
+```
+
+> If you cannot use `mkDerivationWith`, please refer to Locating runtime dependencies below.
```
Comment (jtojnar) [inline doc/languages-frameworks/qt.section.md:32]: This does not look right, it should be multi-paragraph list item and I have no idea where are the blockquotes coming from.

**Review by jtojnar — CHANGES_REQUESTED** [review]

File: doc/languages-frameworks/qt.section.md (hunk 1 above)
Comment (wiltaylor) [reply doc/languages-frameworks/qt.section.md]: Awesome thanks. I was wondering how to get anchors in common markdown, couldn't see it in the spec :).

File: doc/languages-frameworks/qt.section.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+```nix
+mkDerivation {
+  # ...
+
+  # Disable this library with Qt &lt; 5.9.0
+  meta.broken = builtins.compareVersions qtbase.version "5.9.0" &lt; 0;
+}
+```
+**Adding a library to Nixpkgs**
```
Comment (jtojnar) [inline doc/languages-frameworks/qt.section.md:84]: The id should be added to other headings as well.

File: doc/languages-frameworks/qt.section.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
+
+Qt
+=
```
Comment (jtojnar) [inline doc/languages-frameworks/qt.section.md:3]: Use ATX headings for consistency:

```suggestion
# Qt

```

File: doc/languages-frameworks/qt.section.md (hunk 1 above)
Comment (jtojnar) [reply doc/languages-frameworks/qt.section.md]: It is a [pandoc’s markdown extension](https://pandoc.org/MANUAL.html#pandocs-markdown) but I think it makes sense to preserve as much of semantics as possible until actual markdown implementation is selected.

Comment (wiltaylor) [reply doc/languages-frameworks/qt.section.md]: Have updated this.

File: doc/languages-frameworks/qt.section.md (hunk 3 above)
Comment (wiltaylor) [reply doc/languages-frameworks/qt.section.md]: Have made the changes. thanks :)

File: doc/languages-frameworks/qt.section.md (hunk 4 above)
Comment (wiltaylor) [reply doc/languages-frameworks/qt.section.md]: Have updated

File: doc/languages-frameworks/qt.section.md (hunk 2 above)
Comment (wiltaylor) [reply doc/languages-frameworks/qt.section.md:11]: Yeah it is a tricky one. Not sure what we should use instead of them.

Comment (FRidh) [reply doc/languages-frameworks/qt.section.md:11]: In https://github.com/NixOS/nixpkgs/pull/105036 I introduce Sphinx + MyST as CommonMark implementation. That way, we can fall back to rST and have at least a bit more possibilities. Unfortunately, rST does not support callouts, however see https://stackoverflow.com/questions/997428/how-to-write-callouts-using-restructured-text for reStructuredText.

Comment (ryantm) [reply doc/languages-frameworks/qt.section.md:11]: Are we able to stick with the docbook for the callouts for now?

Comment (wiltaylor) [reply doc/languages-frameworks/qt.section.md:11]: Having a bunch of trouble with the docbook tags. pandoc keeps stripping out the outer tags and causes syntax errors with docbook.

I am going to keep looking into it but if anyone has an example of being able to put docbook tags inline I can go off that would be much appreciated.

eg. if I put the following in to the document:

```
### Example buildBowerComponents {#ex-buildBowerComponents}

<programlisting language="nix">
bowerComponents = buildBowerComponents {
  name = "my-web-app";
  generated = ./bower-packages.nix; <co xml:id="ex-buildBowerComponents-1" />
  src = myWebApp; <co xml:id="ex-buildBowerComponents-2" />
};
</programlisting>
```

pandoc generates

```
    <section xml:id="ex-buildBowerComponents">
      <title>Example buildBowerComponents</title>
      bowerComponents = buildBowerComponents { name =
      <quote>my-web-app</quote>; generated = ./bower-packages.nix;
      <co xml:id="ex-buildBowerComponents-1" /> src = myWebApp;
      <co xml:id="ex-buildBowerComponents-2" /> };
      <para>
        In
        <link linkend="ex-buildBowerComponents"><quote>buildBowerComponents</quote>
        example</link> the following arguments are of special
        significance to the function:
      </para>

```

I have tried a bunch of different variations including a whole section but it keeps stripping tags.

Comment (jtojnar) [reply doc/languages-frameworks/qt.section.md:11]: Hmm, I guess that will only work for inline tags :crying_cat_face:

Comment (wiltaylor) [reply doc/languages-frameworks/qt.section.md:11]: Awesome I figured it out.

If you want to put in line docbook stuff and it is stripping it like above you can do the following to stop it stripping the tags:
````
```{=docbook}
docbook markdown here
```
````

Comment (wiltaylor) [reply doc/languages-frameworks/qt.section.md:11]: Added the docbook callouts back in again.

**Review by ryantm — APPROVED** [review]
Diff and side-by-side comparison in web browser look good.

**Comment by ryantm** [comment]
Thanks!

**Pull request merged**